docs/DeveloperGuide.md

```diff
@@ -2,33 +2,166 @@
 
 ## Design & implementation
 
-{Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
+**`HdBuy`** has [`HdBuy`](https://https://github.com/AY2021S2-CS2113-F10-1/tp/blob/master/src/main/java/seedu/hdbuy/HdBuy.java). It is responsible for,
```

Comment:
Should this be further clarified? I am not too sure what  HdBuy has HdBuy means.